src/main/java/database/UserDBAccess.java

```diff
@@ -1,106 +1,55 @@
 package database;
 
-import com.google.gson.Gson;
-import com.google.gson.JsonArray;
-import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import models.UserDBModel;
-import util.RemoteUtilities;
 
-import java.util.ArrayList;
 import java.util.List;
 
-public class UserDBAccess implements UserDBGateway {
-    private final String urlBase;
+public class UserDBAccess extends DBAccess<UserDBModel> implements UserDBGateway {
 
     public UserDBAccess(String urlBase) {
-        this.urlBase = urlBase;
+        super(urlBase);
     }
 
-    /**
-     * Loads the users from the database.
-     * @return a list of UserModel objects representing the existing users
-     */
     @Override
-    public List<UserDBModel> loadUsers() {
-        try {
-            String result = performHttpRequest("GET", null, null);
-
-            // Parse JSON data
-            Gson gson = new Gson();
-            JsonObject jsonObject = gson.fromJson(result, JsonObject.class);
-            JsonArray usersArray = jsonObject.get("users").getAsJsonArray();
-            List<UserDBModel> users = new ArrayList<>();
-            for (JsonElement userElement : usersArray) {
-                JsonObject userObject = userElement.getAsJsonObject();
-                int userId = userObject.get("id").getAsInt();
-                String username = userObject.get("username").getAsString();
-                String password = userObject.get("password").getAsString();
-                users.add(new UserDBModel(userId, username, password));
-            }
+    public List<UserDBModel> retrieveEveryUser() {
```

Comment:
nitpick but I think fetchUsers or getUsers is more standard terminology

Reply:
see f447606